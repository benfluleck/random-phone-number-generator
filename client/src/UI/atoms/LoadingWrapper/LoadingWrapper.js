import React from "react";
import ReactLoading from "react-loading";
import styled from 'styled-components';

const LoadingWrapper = () => (
  <ReactLoading type="spin" color="#1a56f0" height='20%' width='20%' className="loading__wrapper" />
);


export default LoadingWrapper;
