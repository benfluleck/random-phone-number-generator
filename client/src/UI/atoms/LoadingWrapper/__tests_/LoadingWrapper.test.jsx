import React from 'react';
import { shallow } from 'enzyme';

import LoadingWrapper from '../LoadingWrapper';

const setup = (props) => {
  const wrapper = shallow(
    <LoadingWrapper {...props} />
  );

  return {
    wrapper,
  };
};


describe('Component - Loading Wrapper', () => {
  test('should render', () => {
    const { wrapper } = setup();

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
