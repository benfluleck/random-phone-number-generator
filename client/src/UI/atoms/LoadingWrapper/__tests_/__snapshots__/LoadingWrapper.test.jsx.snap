// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Loading Wrapper should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingWrapper />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "loading__wrapper",
      "color": "#1a56f0",
      "delay": 0,
      "height": "20%",
      "type": "spin",
      "width": "20%",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "loading__wrapper",
        "color": "#1a56f0",
        "delay": 0,
        "height": "20%",
        "type": "spin",
        "width": "20%",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
